csrank/choicefunction/baseline.py

```diff
@@ -44,6 +52,11 @@ def predict_scores(self, X, Y, **kwargs):
         """
         self.logger.info("Predicting scores")
 
+        # Its not so straightforward to check if
+        assert (isinstance(X, dict) and isinstance(Y, dict)) or (
+            isinstance(X, np.ndarray) and isinstance(Y, np.ndarray)
+        )
+
```

Comment:
Yes, that would be better. I sort of kept it simple on purpose, since I think the "right" way to do this is to restructure the function in a way that works with `mypy` and then use that.

Then again, "we should do this in the future" is rarely a good strategy so maybe its better to improve the check for now.

Reply:
I have switched it to an exception. Currently the baselines are not tested at all. Do you think I should add a test here?

Reply:
Since this smells a bit like testing something for the sake of testing it, we can leave it as it is right now. We should then do a proper testing of the baselines at a later point in time.